### PR TITLE
UI Improvements

### LINKED ISSUE #732: Tutorial execise icon doesn't turn into in grading icon

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+ Closes #732, #771, #770 

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [ ] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [ ] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [ ] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/utils/CollectionUtil.java [hunk 1]
```diff
@@ -37,11 +37,24 @@ private CollectionUtil() {
...
+   */
+  public static <T, R> List<R> mapWithIndexReverse(@NotNull List<T> list,
+                                                   @NotNull BiFunction<T, Integer, R> func,
+                                                   int startIndex) {
+    int n = list.size();
+    return IntStream
+        .range(0, n)
+        .mapToObj(i -> func.apply(list.get(n - i - 1), n - (startIndex + i) + 1))
+        .collect(Collectors.toList());
+  }
```
Comment (Taikelenn) [inline src/main/java/fi/aalto/cs/apluscourses/utils/CollectionUtil.java:50]: Works nicely, but now the `mapWithIndex` is unused (used only in its own test)
Either:
- remove `mapWithIndex` and change the corresponding test to test `mapWithIndexReverse`
- keep `mapWithIndex` and reverse the list inside ExerciseViewModel.

**Review by Taikelenn — COMMENTED** [review]
Just one tiny code-quality change (functionally it's good)

File: src/main/java/fi/aalto/cs/apluscourses/utils/CollectionUtil.java (hunk 1 above)
Comment (OlliKiljunen) [reply src/main/java/fi/aalto/cs/apluscourses/utils/CollectionUtil.java:50]: So does this method work like (with list `{a,b,c,d}` and `startIndex = 1`):
```
(4,d)
(3,c)
(2,b)
(1,a)
```
In that case, I think it's better to use `mapWithIndex` and then reverse. It's much clearer and less redundant in code. The performance aspect is not important in our case, I think.

**Review by OlliKiljunen — APPROVED** [review]
I agree with Pawel about `mapWithIndexReverse`. Otherwise, this is fine! Well done. :+1:

**Pull request merged**